test/client_test.js

```diff
@@ -1,430 +1,365 @@
+      dummyServer.error({message: 'This is an expected error.'});
     }, 100);
-  }
+  };
 
   var client = new MemJS.Client([dummyServer], {retries: 2});
-  client.set('hello', 'world', function(err, val) {
-    if (err) {
-      errn += 1;
-    }
-    callbn += 1;
-  });
-
-  beforeExit(function() {
-    assert.equal(2, n,  'Ensure set is retried once ' + n);
-    assert.equal(1, callbn,  'Ensure callback is called ' + callbn);
-    assert.equal(1, errn,  'Ensure callback called with error ' + errn);
+  client.set('hello', 'world', function(err /*, val */) {
+    t.equal(2, n, 'Ensure set is retried once');
+    t.ok(err, 'Ensure callback called with error');
+    t.equal('This is an expected error.', err.message);
+    t.end();
   });
-}
+});
 
-exports.testSetErrorConcurrent = function(beforeExit, assert) {
+test('SetErrorConcurrent', function(t) {
   var n = 0;
   var callbn1 = 0;
-  var errn1 = 0;
   var callbn2 = 0;
-  var errn2 = 0;
   var dummyServer = new MemJS.Server();
-  dummyServer.write = function(requestBuf) {
-    setTimeout(function() {
-      request = MemJS.Utils.parseMessage(requestBuf);
+  dummyServer.write = function(/* requestBuf */) {
+    process.nextTick(function() {
       n += 1;
-      dummyServer.error({message: "This is an expected error."});
-    }, 100);
-  }
+      dummyServer.error({message: 'This is an expected error.'});
+    });
+  };
 
   var client = new MemJS.Client([dummyServer], {retries: 2});
-  client.set('hello', 'world', function(err, val) {
-    if (err) {
-      errn1 += 1;
-    }
+  client.set('hello', 'world', function(err /*, val */) {
+    t.ok(err, 'Ensure callback called with error');
+    t.equal('This is an expected error.', err.message);
     callbn1 += 1;
+    done();
   });
 
-  client.set('foo', 'bar', function(err, val) {
-    if (err) {
-      errn2 += 1;
-    }
+  client.set('foo', 'bar', function(err /*, val */) {
+    t.ok(err, 'Ensure callback called with error');
+    t.equal('This is an expected error.', err.message);
     callbn2 += 1;
-  });
-
-  beforeExit(function() {
-    assert.equal(4, n,  'Ensure set is retried once ' + n);
-    assert.equal(1, callbn1,  'Ensure callback is called ' + callbn1);
-    assert.equal(1, errn1,  'Ensure callback called with error ' + errn1);
-    assert.equal(1, callbn2,  'Ensure callback is called ' + callbn2);
-    assert.equal(1, errn2,  'Ensure callback called with error ' + errn2);
-  });
-}
-
-exports.testSetUnicode = function(beforeExit, assert) {
+    done();
+  });
+
+  var done =(function() {
+    var called = 0;
+    return function() {
+      called += 1;
+      if (called < 2) return; 
+      t.equal(2, n,  'Ensure set is retried');
+      t.equal(1, callbn1,  'Ensure callback is called once');
+      t.equal(1, callbn2,  'Ensure callback is called once');
+      process.nextTick(function() {
+        t.equal(4, n,  'Ensure set is retried'); // not sure what's up here..
```

Comment:
Ok we could remove it, but the test called `SetErrorConcurrent` and the fact that there is something unusual happening here makes think having it is a good thing.

I took another look and I think I understand whats happening here and why the value changes on the `nextTick`. In `server.js` when there is an error it gets [sent to all awaiting requests](https://github.com/alevy/memjs/blob/c8d9ff6a581916c6c316b3357c3135e98411bc45/lib/memjs/server.js#L52-L54).  

So when the test queues up two requests and the first one errors, that error goes to both awaiting call backs. Because the clients in this test have `{retries: 2}` each of them is going to re-queue the request and the server is going to send the error, to both, again.

I'm gonna clean up the comment here and verify that after `nextTick` the callbacks are still only called once (which I'm sure is the case). Then this should be good to go.